src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -277,7 +283,15 @@ ReactDOMComponent.Mixin = {
   mountComponent: function(rootID, transaction, context) {
     this._rootNodeID = rootID;
 
-    assertValidProps(this, this._currentElement.props);
+    var props = this._currentElement.props;
+
+    var wrapper = wrappers[this._tag];
+    if (wrapper) {
+      wrapper.mountWrapper.call(this, props, context);
```

Comment:
I would prefer a bunch of switch statements. Avoids the GCC problem, should be faster than a lookup in a map, and avoids the lookup on the wrapper object every time to figure out what mountWrapper means. Might be fast but fragile since it relies on the VM's polymorphic optimizations.